### ISSUE TITLE
Unmet dependencies when installing on Debian testing/bookworm 

### ISSUE THREAD

**Issue by markmandel** [Xpra-org/xpra#3531]
**Describe the bug**

Due to a change in version of python3, as well as some other package changes (see below) xpra will no longer install on Debian testing/bookworm (it actually removes it if you do an `apt full-upgrade`).

**To Reproduce**

```bash
➜  ~ sudo apt install xpra
[sudo] password for mark:
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 xpra : Depends: python3 (< 3.10) but 3.10.4-1 is to be installed
        Depends: libwebp6 (>= 0.5.1) but it is not installable
        Depends: libx264-160 (>= 2:0.160.3011+gitcde9a93) but it is not installable
        Recommends: xpra-html5 but it is not going to be installed
        Recommends: python3-rencode but it is not going to be installed
        Recommends: python3-zeroconf but it is not going to be installed
        Recommends: python3-netifaces but it is not going to be installed
        Recommends: python3-gst-1.0 but it is not going to be installed
        Recommends: gir1.2-appindicator3-0.1 but it is not installable
        Recommends: gir1.2-rsvg-2.0 but it is not going to be installed
        Recommends: ibus but it is not going to be installed
        Recommends: python3-qrencode but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

**System Information (please complete the following information):**
 
```bash
➜  ~ lsb_release --all

No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux bookworm/sid
Release:        testing
Codename:       bookworm
```

Xpra is installed via deb repository:

```bash
➜  sources.list.d cat xpra_org.list
# deb https://xpra.org/ bullseye main
deb https://xpra.org/ bookworm main
```

**Additional context**

I've no idea how relevant it is, but it looks like there are the following similar packages in Debian testing now:

* https://packages.debian.org/bookworm/libx264-163
* https://packages.debian.org/bookworm/libwebp7
* https://packages.debian.org/bookworm/python3.9

Just in case that is useful information.

**Comment by totaam** [comment]
I believe that this is similar to #3525: use the beta channel until bookworm is released.
(as we can't rebuild the stable channel every time Debian breaks the ABI of those packages)

**Comment by markmandel** [comment]
Thanks for the quick response, but unfortunately, this didn't solve the problem.

```bash
➜  cat xpra_org.list
# deb https://xpra.org/ bullseye main
# deb https://xpra.org/ bookworm main
deb [arch=amd64,arm64 signed-by=/usr/share/keyrings/xpra-2022.gpg] https://xpra.org/beta/ bookworm main
```

If I show the package, you can see I'm on `bookworm` and `beta` - but it's still requiring `python3 (<< 3.10)`

```bash
➜  apt show xpra
Package: xpra
Version: 4.4-r31203-1
Priority: optional
Section: x11
Maintainer: Antoine Martin <antoine@xpra.org>
Installed-Size: 13.9 MB
Depends: python3 (<< 3.10), python3 (>= 3.9~), python3:any (>= 3.6~), libavcodec58 (>= 7:4.4), libavformat58 (>= 7:4.4), libavif13 (>= 0.9.3), libavutil56 (>= 7:4.4), libc6 (>= 2.14), libcairo2 (>= 1.2.4), libglib2.0-0 (>= 2.12.0), libgtk-3-0 (>= 3.0.0), libpam0g (>= 0.99.7.1), libswscale5 (>= 7:4.4), libsystemd0, libturbojpeg0 (>= 1:1.4.0), libvpx7 (>= 1.10.0), libwebp7, libx11-6 (>= 2:1.2.99.901), libx264-163 (>= 2:0.163.3060+git5db6aa6), libxcomposite1 (>= 1:0.4.5), libxdamage1 (>= 1:1.1), libxext6, libxfixes3, libxi6 (>= 2:1.2.99.4), libxkbfile1 (>= 1:1.1.0), libxrandr2 (>= 2:1.5.0), libxres1 (>= 2:1.2.1), libxtst6, python3-gi, gir1.2-gtk-3.0, python3-cairo, python3-gi-cairo, python3-opengl, python3-numpy, python3-pil, python3-lz4, python3-brotli, openssl, xvfb, xserver-xorg-video-dummy, keyboard-configuration
Recommends: python3-dbus, xpra-html5, python3-setproctitle, python3-rencode, python3-zeroconf, python3-netifaces, python3-kerberos, python3-gssapi, python3-gst-1.0, python3-cups, python3-paramiko, python3-dnspython, python3-xdg, python3-pyinotify, gir1.2-appindicator3-0.1, gir1.2-rsvg-2.0, dbus-x11, ibus, python3-cryptography, gstreamer1.0-pulseaudio, gstreamer1.0-alsa, gstreamer1.0-plugins-base, gstreamer1.0-tools, gstreamer1.0-plugins-good, gstreamer1.0-plugins-ugly, cups-filters, cups-common, cups-pdf, cups-daemon, openssh-client, sshpass, xterm, python3-qrencode
Suggests: openssh-server, pulseaudio, pulseaudio-utils, v4l2loopback-dkms, python-psutil, gnome-shell-extension-top-icons-plus, gnome-shell-extension-appindicator, pinentry-gnome3, python3-opencv, python3-yaml, python3-pycuda, libnvidia-encode1
Conflicts: python2-xpra, python3-xpra
Homepage: https://xpra.org/
Download-Size: 3,187 kB
APT-Sources: https://xpra.org/beta bookworm/main amd64 Packages
Description: tool to detach/reattach running X programs
 Xpra gives you the functionality of GNU Screen for X applications.
 .
 It allows the user to view remote X applications on their local machine, and
 disconnect and reconnect from the remote machine without losing the state of
 the running applications.
 .
 Unlike VNC, these applications are "rootless".  They appear as individual
 windows inside your window manager rather than being contained within a single
 window.
 .
 This is the Python3 version.

N: There are 27 additional records. Please use the '-a' switch to see them.
```

So it still continues to fail on installation:

```bash
➜  sudo apt install xpra
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 xpra : Depends: python3 (< 3.10) but 3.10.4-1 is to be installed
        Recommends: xpra-html5 but it is not going to be installed
        Recommends: python3-rencode but it is not going to be installed
        Recommends: python3-zeroconf but it is not going to be installed
        Recommends: python3-netifaces but it is not going to be installed
        Recommends: python3-gst-1.0 but it is not going to be installed
        Recommends: gir1.2-appindicator3-0.1 but it is not installable
        Recommends: gir1.2-rsvg-2.0 but it is not going to be installed
        Recommends: ibus but it is not going to be installed
        Recommends: python3-qrencode but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

I did notice in #3525 you mentioned there being periodic releases -- this change to python came at some point this week, so if there is some scheduled release I should wait for that will update based on current dependencies in bookworm, happy to do so.

Thanks!!!

**Comment by totaam** [comment]
Ah, right. The breakage only appeared this week.
Then the latest packages (`xpra_4.4-r31226` landed just now) should use the latest Python - whatever is installed at the time.

(the Debian builds were being held up by a lintian aarch64 bug which was hanging the buildbot: 15463f16ab87bfb0e00c07b2d79d8410d1a19c54)

**Comment by markmandel** [comment]
Looks like that worked! Thanks so much!

Also, thanks for such an amazing project. I live in Xpra day in and day out - it's fantastic.

**Comment by SanjayVas** [comment]
Is this fixed in the latest 4.3 release (v4.3.4), or still only in the beta (4.4) versions?

**Comment by markmandel** [comment]
I've been testing on the Debian testing with  https://xpra.org testing main - and it's working great!

**Comment by totaam** [comment]
Please remember to use the beta channel for unreleased distributions.
We do not rebuild the packages whenever their ABI breaks, the beta channel is rebuilt fairly regularly and usually avoids this problem.

**Comment by gitbxf4** [comment]
Hello, 

I've been unable to install xpra from xpra/repos (I think release or beta) on a Debian unstable machine for a while. I've tried to switch over to building my own xpra, but hitting errors there too. Could someone tell me how to get apt/aptitude to install some recent version of xpra on Debian when I'm getting errors about missing python versions? (It looks like others are, so I think it should work for me too!)

Just like @markmandel  above, I get a message that both a python3 package <3.11 and one that is >=3.10~ is required.

```
The following packages have unmet dependencies:
 xpra : Depends: python3 (< 3.11) but 3.11.1-1 is to be installed
...

> sudo apt-get install python3
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python3 is already the newest version (3.11.1-1).
python3 set to manually installed.
```

My sources.list.d includes:

```
> cat /etc/apt/sources.list.d/xpra-beta.list 
deb [arch=amd64,arm64 signed-by=/usr/share/keyrings/xpra-2022.gpg] https://xpra.org/ sid main
deb [arch=amd64,arm64 signed-by=/usr/share/keyrings/xpra-2022.gpg] https://xpra.org/beta/ sid main
```

Thank you for any thoughts, hints, etc. Take care.

**Comment by totaam** [comment]
> I've been unable to install xpra from xpra/repos (I think release or beta) on a Debian unstable machine for a while

This happens whenever Debian breaks the ABI - which happens quite a lot - just like here:
https://github.com/Xpra-org/xpra/issues/2980#issuecomment-765553765

I have uploaded new beta packages for Debian Bookworm and others this morning - just try again.

> I've tried to switch over to building my own xpra, but hitting errors there too

Please specify these errors so we can fix them.

> My sources.list.d includes:

This looks correct.

**Comment by markmandel** [comment]
Also running beta (thanks for keeping that up to date with breakages, much appreciated!)

This is an old issues, but please feel free to suggest opening a new issue to track, and I'll move this to a new issue.

I had a look at the new build from 2 days ago (27th of Jan) [xpra_5.0-r32584-1_amd64.deb](https://xpra.org/beta/testing/main/binary-amd64/xpra_5.0-r32584-1_amd64.deb) , but when I look at it in apt, I see the same restriction on python packages.

```shell
✦ ❯ apt show xpra
Package: xpra
Version: 5.0-r32584-1
Priority: optional
Section: x11
Maintainer: Antoine Martin <antoine@xpra.org>
Installed-Size: 9,452 kB
Depends: python3 (<< 3.11), python3 (>= 3.10~), python3:any (>= 3.6~), libbrotli1 (>= 0.6.0), libc6 (>= 2.14), libcairo2 (>= 1.2.4), libglib2.0-0 (>= 2.12.0), libgtk-3-0 (>= 3.0.0), liblz4-1 (>= 1.8.2), libpam0g (>= 0.99.7.1), libproc2-0 (>= 2:4.0.2), libqrencode4 (>= 3.3.0), libsystemd0, libx11-6 (>= 2:1.2.99.901), libxcomposite1 (>= 1:0.4.5), libxdamage1 (>= 1:1.1), libxext6, libxfixes3, libxi6 (>= 2:1.2.99.4), libxkbfile1 (>= 1:1.1.0), libxrandr2 (>= 2:1.5.0), libxres1 (>= 2:1.2.1), libxtst6, python3-gi, gir1.2-gtk-3.0, python3-cairo, python3-gi-cairo, python3-opengl, python3-pil, openssl, xvfb, xserver-xorg-video-dummy, keyboard-configuration
....
```

And `python3 (<< 3.11)` is what is causing the issue.

Debian testing/bookwork is currently running

```shell
❯ apt show python3
Package: python3
Version: 3.11.1-1
```

So it fails that check.

As per last time, there could be a build coming down the pipe, in which case, awesome 👍🏻
